src/cargo/core/compiler/build_config.rs

```diff
@@ -70,7 +71,10 @@ impl BuildConfig {
                  its environment, ignoring the `-j` parameter",
             )?;
         }
-        let jobs = jobs.or(cfg.jobs).unwrap_or(::num_cpus::get() as u32);
+        let jobs = match jobs.or(cfg.jobs) {
+            Some(j) => j,
+            None => available_parallelism()?.get() as u32,
```

Comment:
Could the error-propagation here have some context indicating what the error is? (e.g. something like "failed to determine the amount of parallelism available")